verifiers/internal/vsa/verifier.go

```diff
@@ -0,0 +1,201 @@
+package vsa
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+	sigstoreSignature "github.com/sigstore/sigstore/pkg/signature"
+	sigstoreDSSE "github.com/sigstore/sigstore/pkg/signature/dsse"
+	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
+	"github.com/slsa-framework/slsa-verifier/v2/options"
+	vsa10 "github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/vsa/v1.0"
+	"github.com/slsa-framework/slsa-verifier/v2/verifiers/utils"
+)
+
+// VerifyVSA verifies the VSA attestations.
+func VerifyVSA(ctx context.Context,
+	attestations []byte,
+	vsaOpts *options.VSAOpts,
+	verificationOpts *options.VerificationOpts,
+) ([]byte, *utils.TrustedAttesterID, error) {
+	// following steps in https://slsa.dev/spec/v1.1/verification_summary#how-to-verify
+
+	envelope, err := utils.EnvelopeFromBytes(attestations)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// 1. verify the envelope signature,
```

Comment:
nit: These step numbers need an update potentially? (out of order)

Reply:
That's the right order. Step 4 is done by two functions.